### ISSUE TITLE
Fuzzer: make druntime initialization transparent

### ISSUE THREAD

**Issue by JohanEngelen** [ldc-developers/ldc#2225]
To use `-fsanitize=fuzzer`, the user has to manually initialize druntime, because a user-specified `main` is not supported. The user has to supply the following code:
```d
extern (C) int LLVMFuzzerTestOneInput(const(ubyte*) data, size_t size)
{
    // D runtime must be initialized, but only once.
    static bool init = false;
    if (!init)
    {
        import core.runtime : rt_init;
        rt_init();
        init = true;
    }

   //... do the actual fuzz testing
}
```

Perhaps we can add functionality to instead let the user write:
```d
extern (C) int LDC_LLVMFuzzerTestOneInput(const(ubyte*) data, size_t size)
{
   //... do the actual fuzz testing
}
```
where `LLVMFuzzerTestOneInput` initializes druntime and calls `LDC_LLVMFuzzerTestOneInput`.
Another option is to extend libFuzzer functionality or make use of what's already there: https://llvm.org/docs/LibFuzzer.html#startup-initialization

**Comment by JohanEngelen** [comment]
... or with a C module ctor that runs `rt_init()`

**Comment by JohanEngelen** [comment]
resolved by #2510
